### PR TITLE
🚧Add automatic scene updates

### LINKED ISSUE #685: 🚧Add automatic scene updates
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Add a feature to load all scenes in the `Window.SceneManager automatically`.  This should have the ability to auto load enabled by default, but give the user the ability to turn it off to give them control over when and how the scenes are loaded.

Add this same concept to updating, rendering, and unloading

**Do the following:**
1. Auto scene loading:
    - Add a new `bool` property to the `Window` class that can be used to either load or not load the scenes
    - In the `Window.OnLoad()` method, add logic to load all of the scenes if auto-load is turned on
2. Auto scene unloading:
    - Add a new `bool` property to the `Window` class that can be used to either unload or not unload the scenes
    - In the `Window.OnUnload()` method, add logic to load all of the scenes if auto-unload is turned on
3. Auto scene updating
    - Add a new `bool` property to the `Window` class that can be used to either update or not update the scenes
    - In the `Window.OnUpdate()` method, add logic to update all of the scenes if auto-update is turned on
4. Auto scene rendering
    - Add a new `bool` property to the `Window` class that can be used to either draw or not draw the scenes
    - In the `Window.OnDraw()` method, add logic to update all of the scenes if auto-draw is turned on

### Acceptance Criteria

- [x] Auto scene loading added:
    - [x] A new `bool` property has been added to the `Window` class
    - [x] Logic has been added to the `Window.OnLoad()` method
- [x] Auto scene unloading added:
    - [x] A new `bool` property has been added to the `Window` class
    - [x] Logic has been added to the `Window.OnUnload()` method
- [x] Auto scene updating added:
    - [x] A new `bool` property has been added to the `Window` class
    - [x] Logic has been added to the `Window.OnUpdate()` method
- [x] Auto scene rendering added:
    - [x] A new `bool` property has been added to the `Window` class
    - [x] Logic has been added to the `Window.OnDraw()` method
- [x] Create an issue to update the velaptor docs regarding the new auto functionality added to `Window`
    > **Note**
    > This needs to have updates to the guides.  When creating the issue, go through the doc site to find all the guide areas that need updating.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct milestone _(if applicable)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Config Changes      | `⚙️config` |
| Performance Improvements | `🏎️performance` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
### **Description:**

To allow this pull request to be merged, please make sure that the following items of this pull request are complete.

This pull request closes #685

✅The pull request head branch must be a feature branch with the syntax 'feature/<issue-num>-sync-testing'.  
✅The pull request base branch must be a 'main' or 'preview' branch..  
✅The pull request head branch contains a valid issue number.  
✅The pull request title matches the linked issue title exactly.  
✅The pull request assignees match the assignees of the issue.  
✅The pull request labels match the labels of the issue.  
✅The pull request organizational projects match the organizational projects of the issue.  
✅The pull requests milestone matches the milestone of the issue.  

---

### **Additional Info:**
1. Pull requests are automatically synced with the associated issue upon creation by one of the KinsonDigital workflow bots. 
2. The associated issue is the issue number that is embedded in the pull request head branch.
3. The list above will be automatically updated as the pull request's various settings match or do not match the associated issue.
4. To manually sync the pull request to the issue, create a comment with the `[run-sync]` command.
	> **Note** You must be an admin member of the organization to use this command.

---

- [x] Sync with the issue.  Use this to enable or disable this pull request from syncing with its associated issue.

### CODE REVIEW

File: Testing/VelaptorTesting/MainWindow.cs [hunk 1]
```diff
@@ -174,17 +176,18 @@ protected override void OnUpdate(FrameTime frameTime)
     /// <inheritdoc cref="Window.OnDraw"/>
     protected override void OnDraw(FrameTime frameTime)
     {
-        IRenderer.Clear();
-        IRenderer.Begin();
+        base.OnDraw(frameTime);
 
-        SceneManager.Render();
+        // Render the buttons after the 'base.OnDraw()'.  With the rendering being set to auto,
+        // Anymore drawing has to be done after the base.OnDraw() call with the use of
```
Comment (kselena) [inline Testing/VelaptorTesting/MainWindow.cs]: grammar
```suggestion
        // additional drawings have to be done after the base.OnDraw() call with the use of
```

File: Velaptor/OpenGL/GPUData/RectVertexData.cs [hunk 2]
```diff
@@ -8,6 +8,8 @@ namespace Velaptor.OpenGL.GPUData;
 using System.Drawing;
 using System.Numerics;
 
+// TODO: Rename this class
+
```
Comment (KinsonDigitalAdmin) [inline Velaptor/OpenGL/GPUData/RectVertexData.cs]: Do this TODO item

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

File: Testing/VelaptorTests/Fakes/WindowFake.cs [hunk 3]
```diff
@@ -17,9 +18,10 @@ public class WindowFake : Window
     /// </summary>
     /// <param name="window">Mocked window..</param>
```
Comment (kselena) [inline Testing/VelaptorTests/Fakes/WindowFake.cs]: Remove add'l period at the end of the sentence.
```suggestion
    /// <param name="window">Mocked window.</param>
```

File: Velaptor/PushNotifications.cs [hunk 4]
```diff
@@ -89,7 +89,7 @@ internal static class PushNotifications
     /// <summary>
     /// Gets the unique <see cref="Guid"/> for push notifications for when the rectangles need to be rendered.
     /// </summary>
-    public static Guid RenderRectsId { get; } = new ("27c20138-52d3-4b5d-936d-3b62e3b7db4d");
+    public static Guid RenderShapesId { get; } = new ("27c20138-52d3-4b5d-936d-3b62e3b7db4d");
 
     /// <summary>
     /// Gets the unique <see cref="Guid"/> for push notifications for when the lines need to be rendered.
```
Comment (kselena) [inline Velaptor/PushNotifications.cs]: Grammar
```suggestion
    /// Gets the unique <see cref="Guid"/> for push notifications when the lines need to be rendered.
```

File: Velaptor/UI/IWindow.cs [hunk 5]
```diff
@@ -75,15 +75,15 @@ public interface IWindow : IDisposable
     /// <remarks>
     /// <para>
     ///     If this is set to true, this means you do not have to
```
Comment (kselena) [inline Velaptor/UI/IWindow.cs]: SS
```suggestion
    ///     If this is set to true, it means you do not have to
```

File: Velaptor/UI/IWindow.cs [hunk 6]
```diff
@@ -75,15 +75,15 @@ public interface IWindow : IDisposable
...
-    ///     use or invoke the <see cref="IRenderer.Clear"/>() method.
+    ///     use or invoke the <see cref="IBatcher.Clear"/>() method.
     /// </para>
     /// <para>
     ///     Set to the value of <c>false</c> if you want more control over when
```
Comment (kselena) [inline Velaptor/UI/IWindow.cs]: Grammar
```suggestion
    ///     Set to the value of <c>false</c> if you want more control when 
```

File: Velaptor/UI/IWindow.cs [hunk 7]
```diff
@@ -75,15 +75,15 @@ public interface IWindow : IDisposable
...
     ///     the back buffers will be cleared.
     /// </para>
     /// <para>
     ///     WARNING!! - To prevent performance issues, do not have the clear
-    ///     the buffers with the <see cref="IRenderer.Clear"/>() method
+    ///     the buffers with the <see cref="IBatcher.Clear"/>() method
```
Comment (kselena) [inline Velaptor/UI/IWindow.cs]: Grammar
```suggestion
    ///     buffers with the <see cref="IBatcher.Clear"/>() method
```

File: Velaptor/UI/IWindow.cs [hunk 8]
```diff
@@ -75,15 +75,15 @@ public interface IWindow : IDisposable
...
     ///     and set this property to true.  That would be a waste of resources.
```
Comment (kselena) [inline Velaptor/UI/IWindow.cs:87]: SS
```suggestion
    ///     set to  true.  That would be a waste of resources.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Review by kselena — APPROVED** [review]

**Pull request merged**